### ISSUE TITLE
vSphere provider - Network in 'disconnected' state after booting from a template

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-vsphere#26]
_This issue was originally opened by @dkalleg as hashicorp/terraform#6757. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Hi there,
### Terraform Version

Terraform v0.7.0-dev (fae3529da245e66e8df9eadefa0513881f49308d+CHANGES).  
Clean tip of master as of time of posting this.
### Affected Resource(s)
- vSphere provider, vm resource creation
### Terraform Configuration Files

```
resource "vsphere_virtual_machine" "TerraformVm" {

    # Required vars
    name                                = "tfTestVmDisk"
    vcpu                                  = 1
    memory                             = 1024
    datacenter                          = "Datacenter"

    network_interface {
        label = "melody"
    }

    disk {
        template = "ubuntuTemplate"
    }
```
### Debug Output

Provider stuck while waiting for an IP, but is stuck forever since the network does not connect at boot.  If I manually connect the network, the vm will grab an IP and continue on happily and successfully.
### Expected Behavior

Booting from a template, which has a network associated, should connect at boot.
### Actual Behavior

VM boots and is associated with the network, but the network shows in 'disconnected' state and terraform hangs until network is manually connected.
### Steps to Reproduce
1. Create template that has an associated network
2. Create .tf script and define a vm from that template
3. `terraform apply`
4. Watch tf waitForIp forever
### Important Factoids
- Template's network option for "Connect on boot" is enabled.  
- Template's boot option for "Update vmware tools before boot" is enabled.
- Booting a vm from the same template manually will successfully connect to the network at boot (if I select the "Power on after creation" option) through the web ui  

I'm suspecting either some problem with my template, or a problem with the buildNetowrkDevice() function.  I've tinkered with swapping the adapterType between "vmxnet3" and "e1000" with no resolution.  I don't see any additional options we can provide to types.VirtualDeviceConfigSpec / type.VirtualEthernetCard.
Hoping to hear if anyone else even see's this issue.

**Comment by trodemaster** [comment]
I'm also still seeing this issue at unpredictable intervals. My last attempt at getting more details resulted in a huge amount of log output that was not easy to read. Happy to spend some time on this one but need suggestions the approach that will result in useful information.

**Comment by josdotso** [comment]
![screen shot 2017-12-08 at 4 32 21 pm](https://user-images.githubusercontent.com/13571670/33786018-72dad6c2-dc35-11e7-9cfc-b2a25e8e5956.png)

I have this problem as well.  I'm also seeing a problem where cloning and customizing a template on vSAN results in systemd not able to mount root disk inside VM.

**Comment by vancluever** [comment]
This should have been fixed with the 1.0.0 version of this resource as we now only apply incremental changes to network interfaces after cloning - so if a NIC has the correct settings except for network, we don't re-create or otherwise touch the interface other than changing the network that it's connected to. This workflow is the same as it would be if you were to edit a network interface in-place on an already created virtual machine.
